### PR TITLE
Product Gallery: Lock the Sale Badge and the Next/Prev Buttons

### LINKED ISSUE #10868: Product Gallery: Sale badge can be moved, but the change isn't reflected on the page
## Describe the bug

The `Sale Badge` is currently not locked, in the `Product Gallery` block, which means it can be moved.

That said, the change of position is not reflected on the page (frontend).

## To reproduce

1. Using a block theme, go to `Appearance > Editor`.
2. Go to `Templates > Single Product`.
3. Make sure you are using the blockified template.
4. Add the **Product Gallery** block .
5. Move the `Sale Badge` below the `Next/Prev Buttons`.
6. Go to a product page (of a product that's on sale).
7. Notice the position of the `Sale Badge`.

## Expected behavior

I believe the `Sale Badge` should be locked, and unmovable. The block options allow for `left`, `center` and `right` alignment which I believe is all the customization here that should be offered.

## Screenshots

| Editor | Page |
| ------ | ----- |
|Single_Product_‹_Template_‹_Editor_‹_newproductgallery_—_WordPress|V-Neck_T-Shirt_–_newproductgallery|

### PR DESCRIPTION
**Pull request by danieldudzic**
## What

Fixes #10868

- The lock on the `Sale Badge` still allows the user to delete the block.
- For the `Next/Prev Buttons` such thing is not necessary as the user can hide the block with block settings.

## Why

This PR locks the `Sale Badge` and `Next/Prev Buttons` blocks in order to prevent unexpected behavior when these blocks get moved.

## Testing Instructions

_Please consider any edge cases this change may have, and also other areas of the product this may impact._

1. Using a block theme, go to `Appearance > Editor`.
2. Go to `Templates > Single Product`.
3. Make sure you are using the blockified template.
4. Add the **Product Gallery** block .
5. Try to move the `Sale Badge` below the `Next/Prev Buttons`.
6. Notice the blocks are locked.

* [ ] Do not include in the Testing Notes 
* [ ] Should be tested by the development team exclusively 

## Screenshots or screencast

| Before | After |
| ------ | ----- |
|Single_Product_‹_Template_‹_Editor_‹_newproductgallery_—_WordPress-3|Single_Product_‹_Template_‹_Editor_‹_newproductgallery_—_WordPress-2|

## WooCommerce Visibility

Required:

* [ ] WooCommerce Core
* [ ] Feature plugin
* [x] Experimental
* [ ] N/A

## Checklist

Required:
* [x] This PR has either a `[type]` label or a `[skip-changelog]` label.
* [x] This PR is assigned to a milestone.

Conditional:
* [ ] This PR has a changelog description (if `[skip-changelog]` label is not present).
* [ ] This PR adds/removes a feature flag & I've updated [this doc](https://github.com/woocommerce/woocommerce-blocks/blob/trunk/docs/internal-developers/blocks/feature-flags-and-experimental-interfaces.md).
* [ ] This PR adds/removes an experimental interfaces, and I've updated [this doc](https://github.com/woocommerce/woocommerce-blocks/blob/trunk/docs/internal-developers/blocks/feature-flags-and-experimental-interfaces.md).
* [ ] This PR has been accessibility tested.
* [ ] This PR has had any necessary documentation added/updated.

## Changelog

> Product Gallery: Lock the Sale Badge and Next/Prev Buttons.

### CODE REVIEW

File: assets/js/blocks/product-gallery/edit.tsx [hunk 1]
```diff
@@ -71,6 +72,7 @@ const TEMPLATE: InnerBlockTemplate[] = [
 										type: 'flex',
 										verticalAlignment: 'bottom',
 									},
+									lock: { move: true, remove: true },
```
Comment (kmanijak) [inline assets/js/blocks/product-gallery/edit.tsx:74]: Is it expected that the Next/Prev buttons have a removal lock as well? I think it makes sense, but still making sure it's added on purpose.

**Review by kmanijak — APPROVED** [review]
It works fine! Left one question, but pre-approving anyway!

---

Not related to this PR but let me leave a piece of feedback on Next/Prev Buttons.

I noticed the block offers horizontal alignment options that have no influence on the block, so potentially could be removed to avoid confusion.

image

**Comment by danieldudzic** [comment]
> I noticed the block offers horizontal alignment options that have no influence on the block, so potentially could be removed to avoid confusion.

Thanks! This is a known bug, and I have opened an issue for it in Gutenberg a while back:
https://github.com/WordPress/gutenberg/issues/53871

File: assets/js/blocks/product-gallery/edit.tsx (hunk 1 above)
Comment (danieldudzic) [reply assets/js/blocks/product-gallery/edit.tsx:74]: Yes, this is on purpose.

I have tried to explain it in the PR description :)
>- The lock on the Sale Badge still allows the user to delete the block.
>- For the Next/Prev Buttons such thing is not necessary as the user can hide the block with block settings.

**Pull request merged**